File: NaruciBa/NaruciBa.Model/Podkategorija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NaruciBa.Model
{
    public class Podkategorija
    {
        public int PodkategorijaID { get; set; }
        public string Naziv { get; set; }
        public int? KategorijaID { get; set; }
    }
}

File: NaruciBa/naruci_ba_mobile/lib/models/Podkategorija.dart
import 'package:json_annotation/json_annotation.dart';

part 'Podkategorija.g.dart';

@JsonSerializable()
class Podkategorija {
  final int podkategorijaID;
  final String naziv;
  final int kategorijaID;

  Podkategorija({
    required this.podkategorijaID,
    required this.naziv,
    required this.kategorijaID,
  });

  /// Connect the generated [_$PersonFromJson] function to the `fromJson`
  /// factory.
  factory Podkategorija.fromJson(Map<String, dynamic> json) =>
      _$PodkategorijaFromJson(json);

  /// Connect the generated [_$PoslovnicaToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$PodkategorijaToJson(this);
}
